### PR TITLE
Transform `Faucet` from lib to binary

### PR DESCRIPTION
**Pull request by phklive**
In this PR I transform the miden-faucet from a library to a binary. 

The behaviour of the faucet was resembling more a binary than a lib that should be attached to the miden-node hence it's de-coupling.

Closing: #333

### CODE REVIEW

File: bin/node/miden-node.toml
Comment (bobbinth) [inline bin/node/miden-node.toml:1]: Should we remove the `faucet` section from this file?

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left one small comment inline.

File: bin/node/miden-node.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ verify_tx_proofs = true
 
 [rpc]
 # port defined as: sum(ord(c)**p for (p, c) in enumerate('miden-rpc', 1)) % 2**16
-endpoint = { host = "0.0.0.0", port = 57291 }
```
Comment (polydez) [inline bin/node/miden-node.toml:13]: This suppress exposing port to the network in favour of local host connections. Is it was intended for RPC?

File: bin/node/miden-node.toml
Comment (phklive) [reply bin/node/miden-node.toml:1]: Indeed, removed.

File: bin/node/miden-node.toml (hunk 1 above)
Comment (phklive) [reply bin/node/miden-node.toml:13]: These are some default configs for the node. A user can decide to modify them to their will afterwards. Do you believe we should leave the as is?

Comment (polydez) [reply bin/node/miden-node.toml:13]: IMO we should provide defaults which are correct for the most frequent use cases. It's not a big mistake, but I'd prefer to keep it as is.

**Pull request merged**